### PR TITLE
Fix menu images preload

### LINKED ISSUE #32: Index Menu: Background Images on Hover have slight delay
In the Index menu with the project list, when you hover over the project name the background image has a slight delay in loading and doesn't smoothly fade in. Is there a way to fetch this image before it is hovered so that it fades in smoothly?

This is just something I noticed and isn't high-priority at all but would be awesome if we could fix it!

screen shot 2019-01-27 at 2 14 07 pm

So this needs two things done:
- [x] Pre-load background images
- [x] Fix hover gap between project titles

### PR DESCRIPTION
**Pull request by antonlegoo**
Fixes #32

### CODE REVIEW

File: src/app/modules/shared/components/main-menu.vue [hunk 1]
```diff
@@ -151,6 +159,25 @@ export default
 			this.hoverIndexTitle = false;
 		},
 
+		preloadIndexImages ()
+		{
+			// Bail if they're already preloaded
+			if( this.isIndexImagesPreloaded ) return;
+
+			// Prelaod each project's background image
```
Comment (danieltomasku) [inline src/app/modules/shared/components/main-menu.vue]: Small typo here "Prelaod"  👀

File: src/app/modules/shared/components/main-menu.vue [hunk 2]
```diff
@@ -151,6 +159,25 @@ export default
...
+			let images = [];
+			this.indexLinks.forEach( (item, index) => 
+			{
+				// Prealod image
+				images[index] = new Image();
+				images[index].src = item.image;
+			});
+
+			// Set preloaded flag to true
```
Comment (danieltomasku) [inline src/app/modules/shared/components/main-menu.vue]: Same comment here

File: src/app/modules/shared/components/main-menu.vue (hunk 1 above)
Comment (antonlegoo) [reply src/app/modules/shared/components/main-menu.vue]: Fixed!

File: src/app/modules/shared/components/main-menu.vue (hunk 2 above)
Comment (antonlegoo) [reply src/app/modules/shared/components/main-menu.vue]: Fixed!

**Pull request merged**